src/ol/interaction/modifyinteraction.js

```diff
@@ -959,6 +1044,12 @@ ol.interaction.Modify.prototype.insertVertex_ = function(segmentData, vertex) {
       coordinates = geometry.getCoordinates();
       coordinates[depth[0]].splice(index + 1, 0, vertex);
       break;
+    case ol.geom.GeometryType.RECTANGLE:
```

Comment:
Well, I think it would be cleanest if there was an actual object passed from the mouse down handler to the drag handler. This would require some refactoring, but could be used for both the circle and the rectangle.

Currently, the handler would just check if the mouse position snaps to an existing vertex to determine if one is being dragged. Otherwise, the circle handler calculates the nearest point to the circumference and uses that.